File: selection_sort.py
inp = [7, 3, 2, 4, 8, 5, 3, 5, 9, 3]
n = len(inp)



def selection_sort(array, size):
  for i in range(size - 1):
    iMin = i
    for j in range(i + 1, size):
      if array[j] < array[iMin]:
        iMin = j
    temp = array[iMin]
    array[iMin] = array[i]
    array[i] = temp
  return array

print selection_sort(inp, n)

File: selection_sort.cpp
#include <iostream>

using namespace std;
void selection_sort(int A[], int n);
int main(int argc, char const *argv[])
{
  int arr[] = {3, 6, 2, 7, 4, 8};
  int n = 6;
  selection_sort(arr, n);
  for (int i = 0; i < n; ++i)
  {
    cout << arr[i] << " ";
  }
  cout << endl;

  return 0;
}

void selection_sort(int A[], int n) {
  for (int i = 0; i < n - 2; i++)
  {
    int min = i;
    for (int j = i + 1; j < n - 1; j++)
    {
      if(A[j] < A[min]) min = j;
    }
    int temp = A[i];
    A[i] = A[min];
    A[min] = temp;
  }
}